packages/router/src/components/Link.tsx

```diff
@@ -1,49 +1,61 @@
-import * as React from 'react'
-import type { AnchorHTMLAttributes } from 'react'
+import type * as React from 'react'
 import { useInView } from 'react-intersection-observer'
 
 import { useRouter } from '../hooks/useRouter'
 import useRoute from '../hooks/useRoute'
 
-interface TuonoLinkProps {
+interface TuonoLinkProps
+  extends Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, 'preload'> {
   /**
-   * If "true" the route gets loaded when the link enters the viewport. Default "true"
+   * If "true" the route gets loaded when the link enters the viewport.
+   * @default true
    */
   preload?: boolean
   /**
-   * If "false" the scroll offset will be kept across page navigation. Default "true"
+   * If "false" the scroll offset will be kept across page navigation.
+   * @default true
    */
   scroll?: boolean
 }
 
 export default function Link(
-  componentProps: AnchorHTMLAttributes<HTMLAnchorElement> & TuonoLinkProps,
+  componentProps: TuonoLinkProps,
 ): React.JSX.Element {
-  const { preload = true, scroll = true, ...props } = componentProps
+  const {
+    preload = true,
+    scroll = true,
+    children,
+    href,
+    onClick,
+    ...otherProps
```

Comment:
I think `...rest` is more React idiomatic

Reply:
Yo are right